src/script/script_gui.cpp

```diff
@@ -860,11 +860,10 @@ struct ScriptDebugWindow : public Window {
 		/* Detect when the user scrolls the window. Enable autoscroll when the
 		 * bottom-most line becomes visible. */
 		if (this->last_vscroll_pos != this->vscroll->GetPosition()) {
-			this->autoscroll = this->vscroll->GetPosition() >= log.size() - this->vscroll->GetCapacity();
+			this->autoscroll = this->vscroll->GetPosition() + this->vscroll->GetCapacity() >= (int)log.size();
 		}
 		if (this->autoscroll) {
-			int scroll_pos = std::max<int>(0, (int)log.size() - this->vscroll->GetCapacity());
-			if (this->vscroll->SetPosition(scroll_pos)) {
+			if (this->vscroll->SetPosition((int)log.size())) {
```

Comment:
Maybe `SetPosition` should also accept `size_t` like the other ones, as done in the first commit?

Reply:
I thought about that, but that then means the automatic range clamping won't work properly for negative values.